bindings/java/src/org/sleuthkit/datamodel/CaseDbAccessManager.java

```diff
@@ -602,6 +607,49 @@ public void select(final String sql, final CaseDbAccessQueryCallback queryCallba
 		}
 	}
 	
+
+	/**
+	 * Creates a prepared statement object for the purposes of running a select
+	 * statement.
+	 *
+	 * @param sql The select statement without the starting select keyword.
+	 *
+	 * @return The prepared statement object.
+	 *
+	 * @throws TskCoreException
+	 */
+	@Beta
+	public CasePreparedStatement prepareSelect(String sql) throws TskCoreException {
+		String selectSQL = "SELECT " + sql; // NON-NLS
+		try {
+			CaseDbConnection connection = tskDB.getConnection();
+			PreparedStatement statement = connection.getPreparedStatement(selectSQL, Statement.NO_GENERATED_KEYS);
+			return new CasePreparedStatement(connection, statement);
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error creating select prepared statement.", ex);
+		}
+	}
+
+	/**
+	 * Performs a select statement query with the given case prepared statement.
+	 *
+	 * @param preparedStatement The case prepared statement.
+	 * @param queryCallback     The callback to handle the result set.
+	 *
+	 * @throws TskCoreException
+	 */
+	@Beta
+	public void select(CasePreparedStatement preparedStatement, CaseDbAccessQueryCallback queryCallback) throws TskCoreException {
+		tskDB.acquireSingleUserCaseReadLock();
+		try (ResultSet resultSet = preparedStatement.getStatement().executeQuery()) {
+			queryCallback.process(resultSet);
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error running SELECT query.", ex);
```

Comment:
We should make debugging easier by including the SQL that failed to execute in the error message. This might require the CasePreparedStatement to provide a getter for its SQL, but see previous comment where I suggest that we pass the SQL to the constructor for that class.